src/main/java/com/cloudera/director/openstack/OpenStackCredentials.java

```
@@ -6,25 +21,43 @@
 	private final String identity;
 	private final String credential;
 
-	public OpenStackCredentials(String endpoint, String tenant, String user, String credential){
+	public OpenStackCredentials(String endpoint, String tenant, String user, String credential) {
+		String nullArg = "";
+		if (endpoint == null) {
+			nullArg += "The Keystone Endpoint ";
+		}
+		
+		if (tenant == null || user == null) {
+			nullArg += ", Identity(tenant and user) ";
+		}
+		
+		if (credential == null) {
+			nullArg += "and the credential ";
+		}
+		
```

the null checking logic above could generate some strange messages. It would be better to fail fast on null rather than try to have a single message for all the arguments. In our code we follow this pattern:

this.keystone_endpoint = checkNotNull(endpoint, "endpoint is null");